### PR TITLE
[Issue 1975] Add option to require tx replay protection

### LINKED ISSUE #1975: Disable sending of non eip155 replay protected tx by default
Geth recently implemented this change here: https://github.com/ethereum/go-ethereum/pull/22339. It would be nice if Besu would also disable sending of non EIP155 transactions by default (see reasoning in linked PR).

### PR DESCRIPTION
**Pull request by mbaxter**
## PR description
Add new option (`--strict-tx-replay-protection-enabled`) to require locally submitted transaction to use transaction replay protection.  This option defaults to false for now, in the future we should change this to default to true. 

Additional incidental changes include:
* Clean up `Node` interface to remove unnecessary method: https://github.com/hyperledger/besu/pull/2497/commits/18ac1537cdc48a613dd6222d8b4a50edb04e62cf
* Make `GenesisConfigurationFactory` methods static: https://github.com/hyperledger/besu/pull/2497/commits/37bdb8ece89d76487fc0ad8aecafd70a0b3cde5e
* Add new genesis file for testing networks initialized at the London hard fork: https://github.com/hyperledger/besu/pull/2497/commits/ba18529e2a7837294f4f8cb3cad9546b0f2ff562

## Fixed Issue(s)
Fixes #1975

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java [hunk 1]
```diff
@@ -253,6 +263,14 @@ public PendingTransactions getPendingTransactions() {
       return basicValidationResult;
     }
 
+    if (isLocal
+        && configuration.getStrictTransactionReplayProtectionEnabled()
```
Comment (jflo) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:267]: Encapsulating a few of these conditions into a method would make this easier to read. Perhaps we can identify the ones that represent the besu client configuration as enabling replay protection, and group them together?

Comment (mbaxter) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:267]: sure - sounds like a good idea 👍

**Comment by vmichalik** [comment]
@mbaxter is there anything we can do to help you finish off this PR? do you need more reviews still?

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java (hunk 1 above)
Comment (mbaxter) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java]: @jflo - updated and ready for a review if you have time :)

File: besu/src/main/java/org/hyperledger/besu/cli/options/unstable/TransactionPoolOptions.java [hunk 2]
```diff
@@ -33,6 +33,18 @@
   private static final String ETH65_TX_ANNOUNCED_BUFFERING_PERIOD_FLAG =
       "--Xeth65-tx-announced-buffering-period-milliseconds";
 
+  private static final String STRICT_TX_REPLAY_PROTECTION_ENABLED_FLAG =
+      "--strict-tx-replay-protection-enabled";
```
Comment (jflo) [inline besu/src/main/java/org/hyperledger/besu/cli/options/unstable/TransactionPoolOptions.java:37]: Ah, so it IS unstable already. Cool, lets prefix with X.

**Review by jflo — APPROVED** [review]
Approved when unstable option gets an X prefix on it. :)

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java [hunk 3]
```diff
@@ -362,6 +362,9 @@ public void startNode(final BesuNode node) {
     params.add("--auto-log-bloom-caching-enabled");
     params.add("false");
 
+    params.add("--strict-tx-replay-protection-enabled");
```
[the remark this answers is not in the document]
Comment (mbaxter) [reply acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java:365]: It defaults to off (no change from current functionality).  AFAIK, the -X options are usually for internal, dev-related options or temporary / experimental options.  I would be inclined to just release it.

[the remark this answers is not in the document]
Comment (jflo) [reply acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java:365]: Sounds reasonable.

File: CHANGELOG.md [hunk 4]
```diff
@@ -250,6 +250,7 @@ This release contains the activation blocks for London across all supported test
 - Change Ethstats's status from experimental feature to stable. [\#2405](https://github.com/hyperledger/besu/pull/2405)
 - Fixed disabling of native libraries for secp256k1 and altBn128. [\#2163](https://github.com/hyperledger/besu/pull/2163)
 - eth_feeHistory API for wallet providers [\#2466](https://github.com/hyperledger/besu/pull/2466)
+- Add option to require replay protection for locally submitted transactions [\#1975](https://github.com/hyperledger/besu/issues/1975)
```
Comment (macfarla) [inline CHANGELOG.md:253]: we would want this changelog entry in the next future version 22.1.0-RC2 :)

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/condition/eth/ExpectSuccessfulEthSendRawTransaction.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/condition/eth/ExpectSuccessfulEthSendRawTransaction.java:2]: nit: new files should be Copyright contributors to Hyperledger Besu

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/jsonrpc/EthSendRawTransactionTest.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/jsonrpc/EthSendRawTransactionTest.java:2]: nit: new files should be Copyright contributors to Hyperledger Besu

**Review by macfarla — COMMENTED** [review]
couple of nits re copyright but also the changelog entry needs to be moved to the correct version

**Review by jflo — APPROVED** [review]
lgtm

File: CHANGELOG.md (hunk 4 above)
Comment (mbaxter) [reply CHANGELOG.md]: thanks @macfarla - sorry I've been so slow to get back to this! :D

Comment (macfarla) [reply CHANGELOG.md]: no worries - this was an easy fix!

**Pull request merged**